java/code/src/com/suse/manager/xmlrpc/iss/HubHandler.java

```diff
@@ -672,4 +690,88 @@ public int syncPeripheralChannels(User loggedInUser, String fqdn) {
         }
         return 1;
     }
+
+    /**
+     * Migrate the existing ISSv1 slaves to ISSv3 peripherals.
+     *
+     * @param loggedInUser The current user
+     * @param migrationData A list of peripheral migration data
+     * @return the migration result
+     *
+     * @apidoc.doc Migrate the existing ISSv1 slaves to ISSv3 peripherals.
+     * @apidoc.param #session_key()
+     * @apidoc.param
+     *   #array_begin("migration_data")
+     *     #struct_begin("slave_migration_data")
+     *       #prop_desc("string", "fqdn", "The slave fully qualified domain name of the remote server.")
+     *       #prop_desc("string", "token", "The token used to authenticate on the remote server.")
+     *       #prop_desc("string", "root_ca", "The root ca needed to establish a secure connection to the
+     *       remote server.")
+     *     #struct_end()
+     *   #array_end()
+     * @apidoc.returntype $MigrationResultSerializer
+     */
+    public MigrationResult migrateFromISSv1(User loggedInUser, List<Map<String, String>> migrationData) {
+        ensureSatAdmin(loggedInUser);
+
+        if (CollectionUtils.isEmpty(migrationData)) {
+            throw new InvalidParameterException("migration data must not be empty");
+        }
+
+        Map<String, SlaveMigrationData> migrationDataMap;
+
+        try {
+            migrationDataMap = migrationData.stream()
+                .map(SlaveMigrationData::new)
+                .collect(Collectors.toMap(SlaveMigrationData::fqdn, Function.identity()));
+        }
+        catch (NullPointerException ex) {
+            throw new InvalidParameterException("Migration data is not valid: %s".formatted(ex.getMessage()));
+        }
+
+        return migratorFactory.createFor(loggedInUser)
+            .migrateFromV1(migrationDataMap);
+    }
+
+    /**
+     * Migrate the existing ISSv2 slaves to ISSv3 peripherals.
```

Comment:
The work "peripheral" was introduced with ISSv2. We should use it here
```suggestion
     * Migrate the existing ISSv2 peripherals to ISSv3 peripherals.
```